include/aspect/particle/world.h

```diff
@@ -21,729 +21,310 @@
-
-          if (num_send) delete [] num_send;
-          if (num_recv) delete [] num_recv;
-          if (send_offset) delete [] send_offset;
-          if (recv_offset) delete [] recv_offset;
-          if (send_reqs) delete [] send_reqs;
-          if (recv_reqs) delete [] recv_reqs;
+          no_balancing,
+          remove_particles,
+          remove_and_add_particles,
+          repartition
         };
 
         /**
-         * Get the deal.II Mapping associated with this particle world.
+         * Initialize the particle world.
          *
-         * @return The Mapping for this world.
+         * @param [in] new_integrator The new Integrator scheme for this
+         * world.
+         * @param [in] new_manager The new property manager for this
+         * world.
+         * @param [in] max_part_per_cell Threshold for removing
+         * particles from cells. Because the
+         * particle world currently does not declare own input parameters this
+         * is read by the Tracer postprocessor and set in this function.
          */
-        const Mapping<dim> *get_mapping() const
-        {
-          return mapping;
-        };
+        void initialize(Integrator::Interface<dim> *new_integrator,
+                        Interpolator::Interface<dim> *new_interpolator,
+                        Property::Manager<dim> *new_manager,
+                        const ParticleLoadBalancing &load_balancing,
+                        const unsigned int max_part_per_cell,
+                        const unsigned int weight);
 
         /**
-         * Set the deal.II Mapping associated with this particle world.
+         * Get the particle property manager for this particle world.
          *
-         * @param [in] new_mapping The new Mapping for this world.
+         * @param return manager The property manager for this
+         * world.
          */
-        void set_mapping(const Mapping<dim> *new_mapping)
-        {
-          mapping = new_mapping;
-        };
+        const Property::Manager<dim> &
+        get_manager() const;
 
         /**
-         * Set the deal.II Triangulation associated with this particle world
-         * and connects relevant listener for mesh changes.
-         *
-         * @param [in] new_tria The new Triangulation for this world.
+         * Add a particle to this world. If the specified cell does not exist
+         * in the local subdomain an exception will be thrown.
          */
-        void set_triangulation(const parallel::distributed::Triangulation<dim> *new_tria)
-        {
-          //if (triangulation) triangulation->signals.post_refinement.disconnect(std_cxx11::bind(&World::mesh_changed, std_cxx11::ref(*this)));
-          triangulation = new_tria;
-          triangulation->signals.post_refinement.connect(std_cxx11::bind(&World::mesh_changed, std_cxx11::ref(*this)));
-        };
+        void add_particle(const Particle<dim> &particle,
+                          const LevelInd &cell);
```

Comment:
The documentation talks about a specified cell, but the function doesn't take a cell. 

The particle is given as a const reference. What will happen to it when passed to this function?